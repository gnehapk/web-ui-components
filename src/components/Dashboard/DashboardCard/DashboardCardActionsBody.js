import React from 'react';
import PropTypes from 'prop-types';

export const DashboardCardActionsBody = ({ children }) => (
  <div className="kubevirt-dashboard__actions-body">{children}</div>
);

DashboardCardActionsBody.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
