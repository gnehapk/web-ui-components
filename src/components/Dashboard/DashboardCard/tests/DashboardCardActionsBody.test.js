import React from 'react';
import { render } from 'enzyme';

import { DashboardCardActionsBody } from '../DashboardCardActionsBody';
import { Dropdown } from '../../../Form/Dropdown';
import { default as dropdownFixture } from '../../../Form/fixtures/Dropdown.fixture';

const testDashboardCardActionsBody = () => (
  <DashboardCardActionsBody>
    <Dropdown {...dropdownFixture[0].props} />
  </DashboardCardActionsBody>
);

describe('<DashboardCardActionsBody />', () => {
  it('renders correctly', () => {
    const component = render(testDashboardCardActionsBody());
    expect(component).toMatchSnapshot();
  });
});
