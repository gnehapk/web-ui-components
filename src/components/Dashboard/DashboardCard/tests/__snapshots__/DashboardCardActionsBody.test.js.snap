// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardCardActionsBody /> renders correctly 1`] = `
<div
  class="kubevirt-dashboard__actions-body"
>
  <div
    class="btn-group btn-group-justified"
  >
    <div
      class="dropdown btn-group btn-group-default"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="kubevirt-dropdown dropdown-toggle btn btn-default"
        id="1"
        role="button"
        type="button"
      >
        This is dropdown button 
        <span
          class="caret"
        />
      </button>
      <ul
        aria-labelledby="1"
        class="dropdown-menu"
        role="menu"
      >
        <li
          class=""
          role="presentation"
        >
          <a
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            choice1
          </a>
        </li>
        <li
          class=""
          role="presentation"
        >
          <a
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            choice2
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
